codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/ParamsGenerator.java

```diff
@@ -318,6 +359,19 @@ public Void unionShape(UnionShape shape) {
         }
 
         private void defaultComplex(Shape shape) {
+            if (defaultValue.isPresent()) {
+                if (checkRequired) {
+                    writer.write("$1Lparams.key?($2L) ? $3L.build($4L, context: $5L) : $6L",
+                            memberSetter, rubySymbol, symbolProvider.toSymbol(shape).getName(), input, context,
+                            defaultValue.get());
+                } else {
+                    writer.write("$1L(params.key?($2L) ? $3L.build($4L, context: $5L) : $6L)",
+                            memberSetter, rubySymbol, symbolProvider.toSymbol(shape).getName(), input, context,
+                            defaultValue.get());
+                }
```

Comment:
For complex types, like if the default of a map is `{}`, I would think we still want to do param building for that complex type, but the VALUE inside of .build() would be `{}` instead of `params[:map]` or w/e the member is called.

Reply:
fixed